crypto/dh/dh_ameth.c

```diff
@@ -536,6 +536,60 @@ static int dhx_pkey_import_from(const OSSL_PARAM params[], void *vpctx)
     return dh_pkey_import_from_type(params, vpctx, EVP_PKEY_DHX);
 }
 
+#define dh_bn_dup_check(bn) \
+{ \
+  const BIGNUM *f = DH_get0_##bn(dh); \
+  \
+  if (f != NULL && (dupkey->bn = BN_dup(f)) == NULL) \
+      goto err; \
+}
+
+static DH *dh_dup(const DH *dh)
+{
+    DH *dupkey = NULL;
+
+    /* Do not try to duplicate foreign DH keys */
+    if (ossl_dh_get_method(dh) != DH_OpenSSL())
+        return NULL;
+
+    if ((dupkey = ossl_dh_new_ex(dh->libctx)) == NULL)
+        return NULL;
+
+    dupkey->length = DH_get_length(dh);
+    if (!ossl_ffc_params_copy(&dupkey->params, &dh->params))
```

Comment:
It looks like some fields that were added later on have not been copied inside this ossl_ffc_params_copy() function...

    unsigned int flags;
    const char *mdname;
    const char *mdprops;

Reply:
Unless we make those allocated strings I do not think we should copy mdname and mdprops. We can copy flags.